src/macrofier/engine.mjs

```diff
@@ -686,7 +686,7 @@ const convertEnumTemplate = (schema, templateName, templates) => {
   const template = getTemplate(templateName, templates).split('\n')
   for (var i = 0; i < template.length; i++) {
     if (template[i].indexOf('${key}') >= 0) {
-      template[i] = enumSchema.enum.map(value => {
+      template[i] = enumSchema.enum.filter(value => value).map(value => {
```

Comment:
Hmm, we currently don't have any enums w/ `false` or `null` in the values. we might want to map them to strings, though, e.g.: `false` -> `"false"` and `null` -> `"unknown"`.

`"unknown"` is our standard value for enums with an unknown possibility.

Reply:
ok, null is treating like empty line. hence why this change added. let me relook

Reply:
you can see

Reply:
So these two fields need to actually support an unset value. we're using null for that in the JSON-Schema.
What makes sense in Cpp for an unset value?

We could return an error or create a standard enum value, like NONE.

If we filter it out, then the method won't support unset values.

Reply:
Will check about this

Reply:
for cpp there is no Unset values, so just added to set value if there is unset template available. So for javascript this can be used to set unset values. So in this case, for CPP, there is chance to create enum with delimiter for last parameter as well (if last one is null), hence the delimiter handling logic is added after completing the value parsing